tests/ui/format.rs

```diff
@@ -75,4 +75,8 @@ fn main() {
     let _s: String = format!("{}", &*v.join("\n"));
 
     format!("prepend {:+}", "s");
+
+    // Issue #8290
+    let x = "foo";
+    let _ = format!("{x}");
```

Comment:
Could you also add a test were the value is format and where a named argument is used, like this :upside_down_face::

```rs
// Formatted (Ignore)
let y = 10;
format!("{y:+}");

// named argument
format!("{x}", x=y)